Create a Checkstyle module to enforce comment formatting in input files

We haveve noticed inconsistencies in the formatting of single-line comments in the codebase. For instance,  some comments lack a space after the //, making them harder to read. 

Current Behavior:
Comments like `//violation` exist in the codebase without a space after `//`.

Desired Behavior:
All single-line comments should have a space after `//`, like `// violation`.

> I also like space after `//` , and main code base are restricted by some check, we can reuse it if we find it.
> 
> To find it , we can just remove space after // and see what Check is doing violation

I used a `RegexpSinglelineCheck` in `checkstyle-examples-checks.xml`

Below it the example i tested the check:
```
/*xml
<module name="Checker">
  <module name="TreeWalker">
    <module name="AnnotationLocation">
      <property name="allowSamelineSingleParameterlessAnnotation"
                value="false"/>
      <property name="allowSamelineParameterizedAnnotation" value="false"/>
      <property name="allowSamelineMultipleAnnotations" value="true"/>
    </module>
  </module>
</module>


*/

package com.puppycrawl.tools.checkstyle.checks.annotation.annotationlocation;

import org.antlr.v4.runtime.misc.NotNull;
import org.mockito.Mock;

// xdoc section -- start
class Example2 {
  @NotNull
  private boolean field1;
  @Override public int hashCode() { return 1; }
  @NotNull
  private boolean field2;
  @Override
  public boolean equals(Object obj) { return true; }
  @Mock
  DataLoader loader1;
  @SuppressWarnings("deprecation") DataLoader loader;
  @SuppressWarnings("deprecation") public int foo() { return 1; } // ok
  @NotNull @Mock DataLoader loader2;
  //ok above as 'allowSamelineMultipleAnnotations' set to true
}
// xdoc section -- end

```

Result of the build:
```
[ERROR] [checkstyle] [ERROR] /Users/piyushkumarsadangi/Development/new/checkstyle/src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/whitespace/nowhitespacebeforecasedefaultcolon/Example1.java:10: Single-line comment should have a space after //. [ensureSpaceAfterSingleLineComment]
[ERROR] [checkstyle] [ERROR] /Users/piyushkumarsadangi/Development/new/checkstyle/src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespaceafter/Example1.java:10: Single-line comment should have a space after //. [ensureSpaceAfterSingleLineComment]
[ERROR] [checkstyle] [ERROR] /Users/piyushkumarsadangi/Development/new/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationlocation/Example1.java:34: Single-line comment should have a space after //. [ensureSpaceAfterSingleLineComment]
[ERROR] [checkstyle] [ERROR] /Users/piyushkumarsadangi/Development/new/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationlocation/Example2.java:35: Single-line comment should have a space after //. [ensureSpaceAfterSingleLineComment]
[ERROR] [checkstyle] [ERROR] /Users/piyushkumarsadangi/Development/new/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/Example1.java:11: Single-line comment should have a space after //. [ensureSpaceAfterSingleLineComment]
[ERROR] [checkstyle] [ERROR] /Users/piyushkumarsadangi/Development/new/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/Example1.java:12: Single-line comment should have a space after //. [ensureSpaceAfterSingleLineComment]
[ERROR] [checkstyle] [ERROR] /Users/piyushkumarsadangi/Development/new/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/Example1.java:13: Single-line comment should have a space after //. [ensureSpaceAfterSingleLineComment]
[ERROR] [checkstyle] [ERROR] /Users/piyushkumarsadangi/Development/new/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/Example2.java:13: Single-line comment should have a space after //. [ensureSpaceAfterSingleLineComment]
[ERROR] [checkstyle] [ERROR] /Users/piyushkumarsadangi/Development/new/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/Example2.java:14: Single-line comment should have a space after //. [ensureSpaceAfterSingleLineComment]
[ERROR] [checkstyle] [ERROR] /Users/piyushkumarsadangi/Development/new/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/Example2.java:15: Single-line comment should have a space after //. [ensureSpaceAfterSingleLineComment]
[ERROR] [checkstyle] [ERROR] /Users/piyushkumarsadangi/Development/new/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/Example3.java:13: Single-line comment should have a space after //. [ensureSpaceAfterSingleLineComment]
[ERROR] [checkstyle] [ERROR] /Users/piyushkumarsadangi/Development/new/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/Example3.java:14: Single-line comment should have a space after //. [ensureSpaceAfterSingleLineComment]
[ERROR] [checkstyle] [ERROR] /Users/piyushkumarsadangi/Development/new/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/Example3.java:15: Single-line comment should have a space after //. [ensureSpaceAfterSingleLineComment]
[ERROR] [checkstyle] [ERROR] /Users/piyushkumarsadangi/Development/new/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/Example4.java:13: Single-line comment should have a space after //. [ensureSpaceAfterSingleLineComment]
[ERROR] [checkstyle] [ERROR] /Users/piyushkumarsadangi/Development/new/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/Example4.java:14: Single-line comment should have a space after //. [ensureSpaceAfterSingleLineComment]
[ERROR] [checkstyle] [ERROR] /Users/piyushkumarsadangi/Development/new/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/Example4.java:15: Single-line comment should have a space after //. [ensureSpaceAfterSingleLineComment]
[ERROR] [checkstyle] [ERROR] /Users/piyushkumarsadangi/Development/new/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/Example5.java:14: Single-line comment should have a space after //. [ensureSpaceAfterSingleLineComment]
[ERROR] [checkstyle] [ERROR] /Users/piyushkumarsadangi/Development/new/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/Example5.java:15: Single-line comment should have a space after //. [ensureSpaceAfterSingleLineComment]
[ERROR] [checkstyle] [ERROR] /Users/piyushkumarsadangi/Development/new/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/emptylineseparator/Example5.java:16: Single-line comment should have a space after //. [ensureSpaceAfterSingleLineComment]

```
Unfortunately, the check that i added is very primitive and takes everthing into consideration like `///////////////////////////////////////////////////`

--- a/config/checkstyle-examples-checks.xml
+++ b/config/checkstyle-examples-checks.xml
@@ -80,6 +80,12 @@
     <property name="minimum" value="1"/>
     <property name="maximum" value="1"/>
   </module>
+  <module name="RegexpSingleline">
+    <property name="id" value="ensureSpaceAfterSpecificSingleLineComment"/>
+    <property name="format" value="^//(ok|violation)"/>
+    <property name="message" value="There should be a space between '//' and 'ok' or 'violation'."/>
+    <property name="maximum" value="0"/>
+  </module>
 
   <!-- Miscellaneous -->
   <module name="NewlineAtEndOfFile"/>

